### DISCUSSION TITLE
Call for help: Joining focused forces on Postject (resource injector)

### DISCUSSION THREAD

**Discussion by jviotti** [nodejs/single-executable#45] [Announcements]
Hi everybody! As written here: https://github.com/nodejs/single-executable/blob/main/blog/2022-08-05-an-overview-of-the-current-state.md, we have initially broken down the problem of providing SEAs into 3 mostly orthogonal components: the resource injector, the virtual file system and the bootstrapper.

As per our discussions so far, we still need to make more conceptual progress on the VFS and bootstrapper ingredients before we can jump into building a blessed implementation.

Given that the resource injection part does seem conceptually sorted, @dsanders11 @RaisinTen and @robertgzr will focus on getting https://github.com/postmanlabs/postject, Postman's open-source resource injector, production ready. Once that's done, we will at least have shipped a full solution to one of the ingredients!

If you are interested, we would love to on-board you on the Postject project if you would like to help with some coding and enjoy getting deep into binary formats. Here are some suggestions to get you started:

- Look at the issue tracker: https://github.com/postmanlabs/postject/issues
- Reach out to @dsanders11 @RaisinTen or @robertgzr directly (we are friendly!)
- Chat to us on the OpenJS Slack: https://openjs-foundation.slack.com/archives/C03S7Q7RV9R
- Join (some of) us at the [OpenJS Collaborator Summit](https://www.eventbrite.com/e/openjs-foundation-collaborator-summit-dublin-tickets-410575531467) in Dublin on October 2nd

While this is a project originating at Postman, we would love to find ways to co-maintain this with community and potentially put projects like Postject into a open governance model.

cc @arcanis @ovflowd @bnb

**Comment by ovflowd** [comment]
I wish I had more time 👀 but let me see how I can balance time with the upcoming changes for nodejs.dev _and_ postject, sounds interesting enough. Sadly tho, I have no experience with Python, so I don't think I'd be a good fit with the coding there 👀

**Reply by RaisinTen** [reply]
@ovflowd thanks for expressing interest! We are getting rid of the Python parts of the project in https://github.com/postmanlabs/postject/pull/8. We are using the C++ APIs of Lief instead and compiling that to Wasm to get rid of native dependencies. The Cli should be a Js wrapper over the Wasm code.

**Reply by ovflowd** [reply]
I'm also rusty on C++, it's been ages 😅 but anything EcmaScript/Node, count me in :) (Or why not C# 😆)
